src/Microsoft.DotNet.Interactive/Commands/KernelCommand.cs

```diff
@@ -64,4 +134,41 @@ public virtual Task InvokeAsync(KernelInvocationContext context)
 
         return Handler(this, context);
     }
+
+    public void SetId(string id)
+    {
+        _id = id;
+    }
+
+    internal string GetOrCreateId()
+    {
+        if (_id is not null)
+        {
+            return _id;
+        }
+
+        SetId(Guid.NewGuid().ToString("N"));
+
+        return _id;
+    }
+
+    public bool Equals(KernelCommand other)
```

Comment:
consider: It may be a good idea to override `object.Equals` as well (and operators `==` and `!=`)